### ISSUE TITLE
collect CSI on router and save down into USB drive

### ISSUE THREAD

**Issue by yujianyuanhaha** [xieyaxiongfly/Atheros_CSI_tool_OpenWRT_UserSpaceApp_src#8]
Greetings,

One limitation of router is the limited storage ... while the collected CSI file in `.dat` are usually in large size. Is that possible to install a openWRT tool that `ls /mnt/YOUR_DISK/`.

There is a lot of reference teach how to install USB disk tool via `opkg`, but I am concern some extra installtion may conflict with current CSI tool and brick the router. Any one meet the issue can help @xieyaxiongfly 

Best 
Jet

**Comment by bald2718281828** [comment]
Yes.   Do it by building the kernel with the right usb packages.   Don’t try to install stuff with opkg, it often won’t work because the relevant branches have massively diverged.  

I share your concern of brick. usually the TFTP recovery works to unbrick.

Aside from USB, another OK approach is writing the data into sequential files and pulling or pushing the files to server with scripts & scp/sftp .

sincerely,

/eli

**Comment by yujianyuanhaha** [comment]
@bald2718281828  thanks a lot, I just try `opkg` way the router did not brick, but cannot recognize USB either. Building from scratch look like the only way.

Best

Jet
